datasets/colours.py

```diff
@@ -71,10 +83,21 @@ def generate_examples(self, num_examples):
         return examples
 
     def evaluate_correct(
-        self, questions: List[str], responses: List[str], expected_answers: List[str]
+            self, questions: List[str], responses: List[str], expected_answers: List[str]
     ) -> Tuple[int, int, List[str]]:
-        color = expected_answers[0].lower()
-        if color in responses[-1].lower():
-            return 1, 1, [f'"{color}" is in the response.']
-        return 0, 1, [f'"{color}" is NOT in the response.']
+        score = 0
+        max_score = 1
+        response_messages = []
+
+        for expected_color, response_text in zip(expected_answers, responses):
+            result = match_color(response_text, [expected_color])
+            if expected_color == "" and result is False:
+                score += 1
+                response_messages.append("No color expected in the response.")
+            elif result:
+                score += 1
+                response_messages.append(f'"{expected_color}" is in the response.')
+            else:
+                response_messages.append(f'"{expected_color}" is NOT in the response.')
```

Comment:
The scoring function should take care that the score is not greater than one. You can probably just divide it by the number of expected responses.